plugin-flex-ts-template-v2/src/feature-library/custom-transfer-directory/custom-components/ExternalDirectoryTab.tsx

```diff
@@ -50,16 +50,35 @@ const ExternalDirectoryTab = (props: OwnProps) => {
 
   const onTransferEntryClick = (entry: ExternalDirectoryEntry) => async (transferOptions: TransferClickPayload) => {
     const defaultFromNumber = Manager.getInstance().serviceConfiguration.outbound_call_flows.default.caller_id;
-    const callerId = workerAttrs.phone ? workerAttrs.phone : defaultFromNumber;
+    const callerId = workerAttrs.phone
+      ? workerAttrs.phone
+      : workerAttrs.selectedCallerId
+      ? workerAttrs.selectedCallerId
+      : defaultFromNumber;
 
     if (transferOptions.mode === 'WARM')
       Actions.invokeAction('StartExternalWarmTransfer', {
         task: props.task,
         phoneNumber: entry.number,
         callerId,
       });
-    else if (transferOptions.mode === 'COLD')
-      Actions.invokeAction('StartExternalColdTransfer', { task: props.task, phoneNumber: entry.number });
+    else if (transferOptions.mode === 'COLD') {
+      let from;
+      if (
+        (props.task?.attributes?.caller && props.task?.attributes?.caller.startsWith('sip')) ||
```

Comment:
When dialing _to_ sip, the caller ID has fewer restrictions and can be an arbitrary string. The issue that I was solving here is dials _from_ sip, where doing a cold transfer to a non-SIP endpoint would fail without overriding caller ID.